google/datalab/pipeline/commands/_pipeline.py

```diff
@@ -0,0 +1,167 @@
+# Copyright 2015 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+# in compliance with the License. You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under the License
+# is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+# or implied. See the License for the specific language governing permissions and limitations under
+# the License.
+
+"""Google Cloud Platform library - Pipeline IPython Functionality."""
+from __future__ import absolute_import
+from __future__ import print_function
+from __future__ import unicode_literals
+
+try:
+  import IPython
+  import IPython.core.display
+  import IPython.core.magic
+except ImportError:
+  raise Exception('This module can only be deployed in ipython.')
+
+import google.datalab.data
+import google.datalab.utils
+import google.datalab.utils.commands
+
+
+def _create_cell(args, cell_body):
+  """Implements the pipeline cell create magic used to create Pipeline objects.
+
+  The supported syntax is:
+
+      %%pipeline create <args>
+      [<inline YAML>]
+
+  Args:
+    args: the arguments following '%%pipeline create'.
+    cell_body: the contents of the cell
+  """
+  name = args.get('name')
+  if name is None:
+    raise Exception("Pipeline name was not specified.")
+
+  pipeline = google.datalab.pipeline._pipeline.Pipeline(
+      cell_body, name, env=IPython.get_ipython().user_ns)
+  google.datalab.utils.commands.notebook_environment()[name] = pipeline
+
+  return pipeline.py
+
+
+def _create_create_subparser(parser):
```

Comment:
IIUC, you're bringing up two issues:
1. yaml config validation - agreed on this and I've opened b/64951979 to track it.
2. Using add_cell_argument() method to add help strings - agreed on this as well. I've opened b/64952760 to track it.